src/nvim/eval/decode.c

```diff
@@ -285,7 +285,7 @@ typval_T decode_string(const char *const s, const size_t len,
       .v_type = VAR_STRING,
       .v_lock = VAR_UNLOCKED,
       .vval = { .v_string = (char_u *)(
-          s_allocated ? (char *)s : xmemdupz(s, len)) },
+          (s == NULL) || s_allocated ? (char *)s : xmemdupz(s, len)) },
```

Comment:
@ZyX-I should this  set empty string if `s` is null?

Reply:
AFAIR no. Though I would rather write `(s == NULL || s_allocated)` or have no parenthesis (and better first as I do not remember what is the priority of the ternary exactly).